api/src/main/java/org/ccci/idm/user/dao/UserDao.java

```diff
@@ -186,4 +186,14 @@
      * @param group from group
      */
     void removeFromGroup(User user, Group group) throws DaoException;
+
+    /**
+     * Returns all available groups
+     *
+     * Note that this method is not particular to a user, but is temporarily made available here until a
+     * more suitable framework becomes available for providing group dao.
+     *
+     * @return list of all available groups
+     */
+    List<Group> getAllGroups() throws ExceededMaximumAllowedResultsException;
```

Comment:
`throws DaoException`

this allows us to add additional exceptions in the future without modifying the signature.